Template.php

```diff
@@ -1498,24 +1492,21 @@ protected function use_unnamed_params() {
           case "url":
           if ($this->blank($p1)) {
             unset($pAll[0]);
-            $this->param[$param_key]->param = $p1;
-            $this->param[$param_key]->val = implode(" ", $pAll);
+            $this->add_if_new($p1,implode(" ", $pAll));
```

Comment:
Perhaps a check here to evaluate whether `$param_recycled === TRUE` would prevent the same parameter slot being used twice?